backend/src/main/java/dev/k2/commons/MinioXsdFileRepository.java

```diff
@@ -91,4 +91,37 @@ public byte[] getFile(String fileName) {
         }
 
     }
+
+    @Override
+    public void deleteFile(String fileName) {
+
+        try{
+
+            minioClient.removeObject(
+                    RemoveObjectArgs.builder()
+                            .bucket(bucketName)
+                            .object(fileName)
+                            .build()
+            );
+
+        } catch (ErrorResponseException e) {
```

Comment:
Add custom exception(s) (ex. `DeleteFileException`) then intercept that and translate to adequate response using `@ControllerAdvice`.

Reply:
Example: file is not found to be deleted - as we know from REST best practices - we should return 404 response code, and that will be made in @ControllerAdvice.
Another example: file exists but couldn't be deleted - we return 500 response code.
Each of these situations have their own exception that is thrown in repository layer and then intercepted by @ControllerAdvice when response will be created.